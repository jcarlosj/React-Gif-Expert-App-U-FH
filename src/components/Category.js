import React from 'react';

export const Category = ({ category }) => {

    const requestToGiphy = async () => {    //  https://developers.giphy.com/
        const 
            url = `https://api.giphy.com/v1/gifs/search?q=superman&limit=9&api_key=6t2zjzJn3x3LILIFUpF6GKKcR95JTJhK`,
            response = await fetch( url ),
            { data } = await response.json();
        // console.log( data );

        const gifs = data.map( image => {
            return {
                id: image.id,
                title: image.title,
                url: image.images?.downsized_medium.url
            };
        });
        console.log( gifs );
    }
    requestToGiphy();

    return (
        <>
            <h3>{ category }</h3>
        </>
    )
}
