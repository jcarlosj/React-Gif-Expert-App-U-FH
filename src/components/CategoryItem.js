import React from 'react';
import PropTypes from 'prop-types';

export const CategoryItem = ( { title, url } ) => {
    return (
        <div className="card animate__animated animate__bounceIn">
            <img src={ url } alt={ title } title={ title } />        
            <h4>{ title }</h4>
        </div>
   )
}

CategoryItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};