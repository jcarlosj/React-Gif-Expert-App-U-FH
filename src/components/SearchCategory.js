import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const SearchCategory = ({ setCategories }) => {

    const [ inputValue, setInputValue] = useState( '' );

    const handleInputChange = event => {
        // console.log( event.target.value );
        setInputValue( event.target.value );
    }

    const handleSubmit = event => {
        event.preventDefault();

        if( inputValue.trim().length > 2 ) {
            setCategories( categories => [ ...categories, inputValue ] );
            setInputValue( '' );
        }
        
    }

    return (
        <form
            onSubmit={ handleSubmit }
        >
            {/* <pre>{ JSON.stringify( inputValue, null, 4 ) }</pre> */}
            <input 
                type="text" 
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

SearchCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
