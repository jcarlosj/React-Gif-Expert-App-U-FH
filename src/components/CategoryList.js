import React from 'react';

/** Hooks */
import { useFetchGifs } from '../hooks/useFecthGifs';

/** Component */
import { CategoryItem } from './CategoryItem';

/** Functional Components */
export const CategoryList = ({ category }) => {

    const { loading, data:images } = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>

            {   loading && <p className="animate__animated animate__flash">Cargando...</p> }

            <div className="card-grid">
            
                {   images.map( image => (
                        <CategoryItem 
                            key={ image.id }
                            { ...image }    //  Spread
                        />
                    ))
                }
                
            </div>
        </>
    )
}
