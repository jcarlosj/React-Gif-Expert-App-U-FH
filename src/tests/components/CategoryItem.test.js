import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { CategoryItem } from '../../components/CategoryItem';

describe( '<CategoryItem />', () => {

    test( 'debe desplegar correctamente', () => {
        
        const
            image = {
                id: 'ouyy43y4r734',
                title: 'Once Upon a Time',
                url: 'https://media4.giphy.com/media/IBzHGPbQLV5Fm/giphy.gif?cid=3a428325tnk0tt3ae9ee85i0zam2aldtbizw22uyqpllhw15&rid=giphy.gif'
            }, 
            wrapper = shallow(
                <CategoryItem { ...image } />
            );
        // console.log( wrapper.debug() );

        expect( wrapper ).toMatchSnapshot();    //  Verifica que un valor coincida con la instantánea (Snapshop) más reciente

    });

});