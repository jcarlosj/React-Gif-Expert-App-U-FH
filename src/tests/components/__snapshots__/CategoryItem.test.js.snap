// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryItem /> debe desplegar correctamente 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Once Upon a Time"
    src="https://media4.giphy.com/media/IBzHGPbQLV5Fm/giphy.gif?cid=3a428325tnk0tt3ae9ee85i0zam2aldtbizw22uyqpllhw15&rid=giphy.gif"
    title="Once Upon a Time"
  />
  <h4>
    Once Upon a Time
  </h4>
</div>
`;
